src/diff_walker.rs

```diff
@@ -14,6 +14,81 @@ pub struct DiffWalker {
     pub rhs_root: RootSchema,
 }
 
+trait DiffScore {
+    fn diff_score(&mut self, rhs: &mut Self) -> usize;
+}
+
+impl DiffScore for Schema {
+    fn diff_score(&mut self, rhs: &mut Self) -> usize {
+        self.clone()
+            .into_object()
+            .diff_score(&mut rhs.clone().into_object())
+    }
+}
+impl DiffScore for SchemaObject {
```

Comment:
actually! why not invoke the main `diff` function as a way to compute the score?

Reply:
nice idea